From 02f11c698d1b12225d9042e9bba5b0ee9be03889 Mon Sep 17 00:00:00 2001
From: Richard Laager <rlaager@wiktel.com>
Date: Sun, 3 Feb 2019 20:18:04 -0600
Subject: [PATCH] Fix a typo

---
 libntp/authreadkeys.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libntp/authreadkeys.c b/libntp/authreadkeys.c
index c304255c4..953325e0a 100644
--- a/libntp/authreadkeys.c
+++ b/libntp/authreadkeys.c
@@ -358,7 +358,7 @@ msyslog(LOG_ERR, "AUTH: authreadkeys: reading %s", file);
 			if ((2*sizeof(keystr)) < jlim) {
 			  jlim =  2 * sizeof(keystr);
 			  msyslog(LOG_ERR,
-			    "AUTH: authreadkeys: key %u trucated to %u bytes",
+			    "AUTH: authreadkeys: key %u truncated to %u bytes",
 			    keyno, (unsigned int)jlim);
 
 			}
-- 
2.17.1

