From 59f063b65ba1ac7d75fd51298dd31bae3c14252c Mon Sep 17 00:00:00 2001
From: Richard Laager <rlaager@wiktel.com>
Date: Sun, 3 Feb 2019 23:25:03 -0600
Subject: [PATCH] Add Documentation= to ntp-wait.service

---
 etc/ntp-wait.service | 1 +
 1 file changed, 1 insertion(+)

diff --git a/etc/ntp-wait.service b/etc/ntp-wait.service
index 66c7bc299..e0bfef2b8 100644
--- a/etc/ntp-wait.service
+++ b/etc/ntp-wait.service
@@ -1,5 +1,6 @@
 [Unit]
 Description=Wait for ntpd to synchronize system clock
+Documentation=man:ntpwait(8)
 Requisite=ntpd.service
 After=ntpd.service
 Conflicts=systemd-timesyncd.service
-- 
2.17.1

