From 9374e15720a79c71e017ba9851d20c890cbe6e2d Mon Sep 17 00:00:00 2001
From: Mark Atwood <mark.atwood@ntpsec.org>
Date: Wed, 23 Nov 2016 22:07:36 -0800
Subject: [PATCH 001/268] version 0.9.5

Signed-off-by: Mark Atwood <mark.atwood@ntpsec.org>
---
 NEWS    | 2 +-
 VERSION | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/NEWS b/NEWS
index 54d1f71..60915c4 100644
--- a/NEWS
+++ b/NEWS
@@ -6,7 +6,7 @@ Much of the traditional function of a news file is now better addressed
 by browsing the comments in the revision history.  This file will focus
 on user-visible changes.
 
-== Repository head ==
+== 2016-11-23: 0.9.5 ==
 
 This release includes a substantial refactoring of the core protocol
 implementation. Due to unresolvable security issues, support for
diff --git a/VERSION b/VERSION
index b0bb878..85b7c69 100644
--- a/VERSION
+++ b/VERSION
@@ -1 +1 @@
-0.9.5
+0.9.6
-- 
2.7.4

