From a0d50605127a6be8d39657b3339114aa22efc53a Mon Sep 17 00:00:00 2001
From: Matt Selsky <matthew.selsky@twosigma.com>
Date: Sun, 3 Feb 2019 20:12:56 -0500
Subject: [PATCH] Make sure PEP241 egg gets variables substituted

---
 pylib/wscript | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pylib/wscript b/pylib/wscript
index be933afc7..0181f8afe 100644
--- a/pylib/wscript
+++ b/pylib/wscript
@@ -83,7 +83,6 @@ def build(ctx):
     egg = ['ntp-%s.info' % ctx.env.NTPSEC_VERSION]
     ctx(
         features="subst",
-        is_copy=True,
         source=['ntp-in.info'],
         target=egg
     )
-- 
2.17.1

