From 83b462b3b3279fc2b2f87e1d5e002d2ff24b1918 Mon Sep 17 00:00:00 2001
From: Richard Laager <rlaager@wiktel.com>
Date: Sat, 10 Dec 2016 22:48:59 -0600
Subject: [PATCH 3/8] Fix missing sntp to ntpdig renames

---
 ntpclients/ntpdig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ntpclients/ntpdig b/ntpclients/ntpdig
index 68088e3..9476834 100755
--- a/ntpclients/ntpdig
+++ b/ntpclients/ntpdig
@@ -213,7 +213,7 @@ def report(packet, json):
         say(" s%d %s\n" % (packet.stratum, packet.leap()))
 
 usage = """
-USAGE:  sntp [-<flag> [<val>] | --<name>[{=| }<val>]]...
+USAGE:  ntpdig [-<flag> [<val>] | --<name>[{=| }<val>]]...
 		[ hostname-or-IP ...]
   Flg Arg Option-Name     Description
    -4 no  ipv4           Force IPv4 DNS name resolution
@@ -234,7 +234,7 @@ USAGE:  sntp [-<flag> [<val>] | --<name>[{=| }<val>]]...
    -s no  slew            Set (slew) the time with adjtime()
 				 - prohibits the option 'slew'
    -t Num timeout         Request timeout in seconds (default 5)
-   -k Str keyfile         Specify a keyfile. SNTP will look in this file
+   -k Str keyfile         Specify a keyfile. ntpdig will look in this file
                           for the key specified with -a
    -V no version          Output version information and exit
    -h no  help            Display extended usage information and exit
-- 
2.7.4

