From 001875132a8e788b61f47481c81dc03dcd47863c Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Thu, 24 Nov 2016 12:24:13 -0500
Subject: [PATCH 007/268] Record the ntpkeygen move.

---
 NEWS | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/NEWS b/NEWS
index 7e69b4a..9661ea6 100644
--- a/NEWS
+++ b/NEWS
@@ -8,6 +8,10 @@ on user-visible changes.
 
 == Repository head ==
 
+ntpkeygen has been moved from C to Python.  This is not a functional
+change, just another move to improve maintainability and reduce attack
+surface by decreasing line count.
+
 == 2016-11-23: 0.9.5 ==
 
 This release includes a substantial refactoring of the core protocol
-- 
2.7.4

