From 115f94a9b28b63c850b0d7c64b662a4c44279cdc Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Thu, 24 Nov 2016 17:51:39 -0500
Subject: [PATCH 008/268] Fix a minor g;itch reported by Hal.

---
 ntpq/ntpq | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ntpq/ntpq b/ntpq/ntpq
index f06e694..da3bd84 100755
--- a/ntpq/ntpq
+++ b/ntpq/ntpq
@@ -943,7 +943,7 @@ usage: writelist [ assocID ]
         associd = self.__assoc_valid(line)
         if associd >= 0:
             qtype = ntp.ntpc.TYPE_SYS if associd == 0 else ntp.ntpc.TYPE_PEER
-            self.__dolist(line.split()[1:], associd, ntp.packet.CTL_OP_READVAR, qtype)
+            self.__dolist(line.split()[1:], associd, ntp.packet.CTL_OP_READVAR, qtype, quiet=True)
 
     def help_readvar(self):
         self.say("""\
-- 
2.7.4

