From 40b917a965178ebeac05c3682c03f8dbfff6e969 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Fri, 25 Nov 2016 09:46:57 -0500
Subject: [PATCH 012/268] Correct reversal of sort order in MRU list.

---
 pylib/packet.py | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/pylib/packet.py b/pylib/packet.py
index 4969272..b71b54d 100644
--- a/pylib/packet.py
+++ b/pylib/packet.py
@@ -1137,16 +1137,18 @@ class ControlSession:
             if "sort" in variables:
                 sorter = variables["sort"]
                 del variables["sort"]
+                # Earliest elements in a span get displayed first, so these
+                # sorts are all sign-reversed from what you might expect.
                 # FIXME: implement sorting by address, in case anyone cares
                 sortdict = {
-                "lstint" : lambda e: e.last,		# lstint ascending
-                "-lstint" : lambda e: -e.last,		# lstint descending
-                "avgint" : lambda e: e.avgint(),	# avgint ascending
-                "-avgint" : lambda e: -e.avgint(),	# avgint descending
+                "lstint" : lambda e: -e.last,		# lstint ascending
+                "-lstint" : lambda e: e.last,		# lstint descending
+                "avgint" : lambda e: -e.avgint(),	# avgint ascending
+                "-avgint" : lambda e: e.avgint(),	# avgint descending
                 "addr" : None,			# IPv4 asc. then IPv6 asc.
                 "-addr" : None,			# IPv6 desc. then IPv4 desc.
-                "count" : lambda e: e.ct,	# hit count ascending
-                "-count": lambda e: -e.ct,	# hit count descending
+                "count" : lambda e: -e.ct,	# hit count ascending
+                "-count": lambda e: e.ct,	# hit count descending
                 }
                 if sorter == "listint":
                     sorter = None
-- 
2.7.4

