From 5e5f8837a5324244513a661f9de033c558ab8e0c Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Sat, 26 Nov 2016 14:53:02 -0500
Subject: [PATCH 019/268] Cope with commas inside strings in control responses.

---
 pylib/packet.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/pylib/packet.py b/pylib/packet.py
index 9e0f409..9a302fa 100644
--- a/pylib/packet.py
+++ b/pylib/packet.py
@@ -1102,12 +1102,18 @@ class ControlSession:
         response = response.rstrip()
         items = []
         if response:
+            instring = False
+            for i in range(len(response)):
+                if response[i] == '"':
+                    instring = not instring
+                elif instring and response[i] == ',':
+                    response[i] = "\xae"
             for pair in response.split(","):
                 try:
                     pair = pair.strip()
                     eq = pair.index("=")
                     var = pair[:eq].strip()
-                    val = pair[eq+1:].strip()
+                    val = pair[eq+1:].strip().replace("\xae", ",")
                     try:
                         val = int(val, 0)
                     except ValueError:
-- 
2.7.4

