From a97e0b13ef5d7fa477f389e67068e582c99fcb7f Mon Sep 17 00:00:00 2001
From: Matt Selsky <matthew.selsky@twosigma.com>
Date: Sat, 26 Nov 2016 11:25:57 -0500
Subject: [PATCH 025/268] refid_as_address() is not static, so make sure to use
 "self"

---
 pylib/packet.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pylib/packet.py b/pylib/packet.py
index 02510ca..107cc42 100644
--- a/pylib/packet.py
+++ b/pylib/packet.py
@@ -489,9 +489,9 @@ class SyncPacket(Packet):
         "Represent a posixized sync packet in an eyeball-friendly format."
         r = "<NTP:%s:%d%:%d" % (self.leap(), self.version(), self.mode())
         r += "%f:%f:" % (self.root_delay, self.root_dispersion)
-        rs = self.refid_as_string
+        rs = self.refid_as_string()
         if not rs.isprint():
-            rd = refid_as_address()
+            rd = self.refid_as_address()
         r += ":" + rs 
         r += ":" + ntp.util.rfc3339(SyncPacket.ntp_to_posix(self.reference_timestamp))
         r += ":" + ntp.util.rfc3339(SyncPacket.ntp_to_posix(self.origin_timestamp))
-- 
2.7.4

