From 59f98cb3c9abbef914b7de1837f70924e6b854c9 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Sat, 26 Nov 2016 23:57:00 -0500
Subject: [PATCH 034/268] Addrrss GitLab issue #165: ntpq -p fails on Python3

---
 pylib/packet.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pylib/packet.py b/pylib/packet.py
index bc44463..509c3d5 100644
--- a/pylib/packet.py
+++ b/pylib/packet.py
@@ -1048,7 +1048,7 @@ class ControlSession:
                 else:
                     if self.debug:
                         warn("Fragment collection ends\n")
-                    self.response = "".join([frag.data for frag in fragments])
+                    self.response = polybytes("".join([polystr(frag.data) for frag in fragments]))
                     if self.debug >= 4:
                         sys.stdout.write("Response packet:\n")
                         dump_hex_printable(self.response)
-- 
2.7.4

