From 77c8a205820877caa901a08738c118fc5fa09d68 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Sun, 27 Nov 2016 04:50:17 -0500
Subject: [PATCH 038/268] Pass baudrate into the refclock_open() function
 correctly.

---
 ntpd/refclock_nmea.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ntpd/refclock_nmea.c b/ntpd/refclock_nmea.c
index 623ec6a..a0936f3 100644
--- a/ntpd/refclock_nmea.c
+++ b/ntpd/refclock_nmea.c
@@ -475,7 +475,7 @@ nmea_start(
 	/* Open serial port. Use CLK line discipline, if available. */
 	snprintf(device, sizeof(device), DEVICE, unit);
 	pp->io.fd = refclock_open(peer->path ? peer->path : device,
-				  peer->baud ? peer->baud : baudrate,
+				  baudrate,
 				  LDISC_CLK);
 	if (0 >= pp->io.fd) {
 		pp->io.fd = nmead_open(device);
-- 
2.7.4

