From 1248f3c872cc043243179590f1a66bbbff383f3d Mon Sep 17 00:00:00 2001
From: Achim Gratz <Stromeko@nexgo.de>
Date: Sun, 27 Nov 2016 07:04:50 -0500
Subject: [PATCH 041/268] Make cv and firiends use the terminal width.

---
 ntpq/ntpq | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ntpq/ntpq b/ntpq/ntpq
index da3bd84..9a3f130 100755
--- a/ntpq/ntpq
+++ b/ntpq/ntpq
@@ -443,7 +443,7 @@ usage: help [ command ]
                         lastcount = 0
                     else:
                         lastcount += 1
-                if lastcount + len(item) > 78:
+                if lastcount + len(item) > self.termwidth - 2:
                     text = text[:-1] + "\n"
                 text += item
             text = text[:-2] + "\n"
-- 
2.7.4

