From c422df5f12ef2864afdc44dd7c12e17d76782c99 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Sun, 27 Nov 2016 07:46:43 -0500
Subject: [PATCH 043/268] Fix for "'str' object does not support item
 assignment" error.

---
 pylib/packet.py | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/pylib/packet.py b/pylib/packet.py
index 043e24f..1f2ae76 100644
--- a/pylib/packet.py
+++ b/pylib/packet.py
@@ -1104,19 +1104,19 @@ class ControlSession:
         # Strip out NULs and binary garbage from text;
         # ntpd seems prone to generate these, especially
         # in reslist responses.
+        instring = False
         response = ""
         for c in self.response:
-            if polyord(c) > 0 and polyord(c) < 127:
-                response += c
+            if c == '"':
+                response = response + c
+                instring = not instring
+            if instring and c == ',':
+                response = response + "\xae"
+            elif polyord(c) > 0 and polyord(c) < 127:
+                response = response + c
         response = response.rstrip()
         items = []
         if response:
-            instring = False
-            for i in range(len(response)):
-                if response[i] == '"':
-                    instring = not instring
-                elif instring and response[i] == ',':
-                    response[i] = "\xae"
             for pair in response.split(","):
                 try:
                     pair = pair.strip()
-- 
2.7.4

