From bea67d8f23c3981dbe661d3df9bbd8c9880e783b Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Mon, 28 Nov 2016 10:14:46 -0500
Subject: [PATCH 053/268] Suppress /bin/stty error message thrown by an ssh
 session.

---
 pylib/util.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pylib/util.py b/pylib/util.py
index a530463..cbd1da6 100644
--- a/pylib/util.py
+++ b/pylib/util.py
@@ -67,7 +67,7 @@ def termsize():
     # Alternatives at http://stackoverflow.com/questions/566746/how-to-get-console-window-width-in-python
     import shlex, subprocess, re
     try:
-        output = subprocess.check_output(shlex.split('/bin/stty -a'), universal_newlines=True)
+        output = subprocess.check_output(shlex.split('/bin/stty -a 2>/dev/null'), universal_newlines=True)
     except (OSError, subprocess.CalledProcessError, AttributeError):
         return (24, 80)
     for pattern in ('rows\D+(\d+); columns\D+(\d+);', '\s+(\d+)\D+rows;\s+(\d+)\D+columns;'):
-- 
2.7.4

