From 099ced205a66cdd7ae46e711cb8a4885d33b8946 Mon Sep 17 00:00:00 2001
From: Matt Selsky <matthew.selsky@twosigma.com>
Date: Sun, 27 Nov 2016 22:50:54 -0500
Subject: [PATCH 054/268] Make sure to pass a keytype to the error messages

---
 ntpq/ntpq | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ntpq/ntpq b/ntpq/ntpq
index 123355d..52f66a8 100755
--- a/ntpq/ntpq
+++ b/ntpq/ntpq
@@ -800,9 +800,9 @@ usage: ntpversion [ version number ]
         if not line:
             self.say("Keytype: %s\n" % self.session.keytype)
         elif not line in "DSA, DSA-SHA, MD4, MD5, MDC2, RIPEMD160, SHA, SHA1":
-            self.warn("Keytype %s is not supported by ntpd.")
+            self.warn("Keytype %s is not supported by ntpd.\n" % line)
         elif line not in hashlib.algorithms_available:
-            self.warn("Keytype %s is not supported by ntpq.")
+            self.warn("Keytype %s is not supported by ntpq.\n" % line)
         else:
             self.session.keytype = line
 
-- 
2.7.4

