From b9af551b537fd9212d10c76c404c73d8b8da86f7 Mon Sep 17 00:00:00 2001
From: Matt Selsky <matthew.selsky@twosigma.com>
Date: Sun, 27 Nov 2016 23:14:57 -0500
Subject: [PATCH 055/268] Fix syntax error for format string

---
 ntpq/ntpq | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ntpq/ntpq b/ntpq/ntpq
index 52f66a8..514811b 100755
--- a/ntpq/ntpq
+++ b/ntpq/ntpq
@@ -338,7 +338,7 @@ usage: help [ command ]
                 return
             if not variables:
                 if len(self.chosts) > 1:
-                    self.warn("server=%s ", self.session.hostname)
+                    self.warn("server=%s " % self.session.hostname)
                 self.warn("***No information returned for association %d\n" \
                                  % associd)
                 continue
-- 
2.7.4

