From b973b1fdb9d9d15b9cd7115e834653f5a961e0be Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Thu, 1 Dec 2016 16:32:28 -0500
Subject: [PATCH 092/268] A safer version of termsize, and referencing it.

---
 ntpq/ntpq     |  7 ++++---
 pylib/util.py | 21 ++++++++++++---------
 2 files changed, 16 insertions(+), 12 deletions(-)

diff --git a/ntpq/ntpq b/ntpq/ntpq
index 91a607a..70323f0 100755
--- a/ntpq/ntpq
+++ b/ntpq/ntpq
@@ -341,9 +341,10 @@ usage: help [ command ]
         if not self.__dogetassoc():
             return
         if self.showhostnames:
-            termwidth = ntp.util.termsize()[1]
+            termwidth = ntp.util.termsize().width
         else:
-            termwidth = None	# Default width 
+            termwidth = None	# Default width
+        print("Width: %d" % termwidth)
         report = ntp.util.PeerSummary(mode,
                                       self.pktversion,
                                       self.showhostnames,
@@ -496,7 +497,7 @@ usage: help [ command ]
                         lastcount = 0
                     else:
                         lastcount += 1
-                if lastcount + len(item) > ntp.util.termsize()[1] - 2:
+                if lastcount + len(item) > ntp.util.termsize().width - 2:
                     text = text[:-1] + "\n"
                 text += item
             text = text[:-2] + "\n"
diff --git a/pylib/util.py b/pylib/util.py
index 4fed333..a2c9999 100644
--- a/pylib/util.py
+++ b/pylib/util.py
@@ -9,6 +9,7 @@ import time
 import os
 import re
 import shutil
+import collections
 
 import ntp.ntpc
 import ntp.version
@@ -63,21 +64,23 @@ def canonicalize_dns(hostname, family=socket.AF_UNSPEC):
         return canonicalized.lower() + portsuffix
     return name[0].lower() + portsuffix
 
+TermSize = collections.namedtuple("TermSize", ["width", "height"])
+
 def termsize():
     "Return the current terminal size."
     # Alternatives at http://stackoverflow.com/questions/566746/how-to-get-console-window-width-in-python
     if not os.isatty(1):
-        return (80, 24)
+        size = (80, 24)
     try:
-        return shutil.get_terminal_size((80, 24))
+        size = shutil.get_terminal_size((80, 24))
     except AttributeError:
-        pass
-    # OK, Python version < 3.3, cope
-    import fcntl, termios, struct
-    h, w, hp, wp = struct.unpack('HHHH',
-        fcntl.ioctl(2, termios.TIOCGWINSZ,
-        struct.pack('HHHH', 0, 0, 0, 0)))
-    return w, h
+        # OK, Python version < 3.3, cope
+        import fcntl, termios, struct
+        h, w, hp, wp = struct.unpack('HHHH',
+            fcntl.ioctl(2, termios.TIOCGWINSZ,
+            struct.pack('HHHH', 0, 0, 0, 0)))
+        size = (w, h)
+    return TermSize(w, h)
 
 class PeerSummary:
     "Reusable report generator for peer statistics"
-- 
2.7.4

