From 914f73217f413e62c6ba1006862b8caa045e6258 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Thu, 1 Dec 2016 17:56:04 -0500
Subject: [PATCH 093/268] Address GitLab issue #178: "ntpq -pn" broken

---
 ntpq/ntpq | 1 -
 1 file changed, 1 deletion(-)

diff --git a/ntpq/ntpq b/ntpq/ntpq
index 70323f0..2b82f2c 100755
--- a/ntpq/ntpq
+++ b/ntpq/ntpq
@@ -344,7 +344,6 @@ usage: help [ command ]
             termwidth = ntp.util.termsize().width
         else:
             termwidth = None	# Default width
-        print("Width: %d" % termwidth)
         report = ntp.util.PeerSummary(mode,
                                       self.pktversion,
                                       self.showhostnames,
-- 
2.7.4

