From 028c2507b91053afb7f5c4465065ce190bd39134 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Thu, 1 Dec 2016 18:29:10 -0500
Subject: [PATCH 094/268] Address Gitlab issue #177: ntpq > TypeError in
 packet.py

---
 pylib/packet.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/pylib/packet.py b/pylib/packet.py
index bfdfaf6..f9905fb 100644
--- a/pylib/packet.py
+++ b/pylib/packet.py
@@ -373,7 +373,7 @@ class SyncPacket(Packet):
         self.origin_timestamp = 0
         self.receive_timestamp = 0
         self.transmit_timestamp = 0
-        self.data = data
+        self.data = polybyes(data)
         self.extension = ''
         self.extfields = []
         self.mac = ''
@@ -567,6 +567,7 @@ class ControlPacket(Packet):
         return "%5d %5d\t%3d octets\n" % (self.offset, self.end(), self.count)
 
     def analyze(self, rawdata):
+        rawdata = polybytes(rawdata)        
         (self.li_vn_mode,
          self.r_e_m_op,
          self.sequence,
-- 
2.7.4

