From 5a16aed05afc06d7f46b820615e46af18cafc2a5 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Thu, 1 Dec 2016 21:25:00 -0500
Subject: [PATCH 096/268] Address GitLab issue #180: Build failure on
 commit/31f0296a

---
 pylib/packet.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/pylib/packet.py b/pylib/packet.py
index 5b492fb..8c5e7f7 100644
--- a/pylib/packet.py
+++ b/pylib/packet.py
@@ -792,14 +792,14 @@ class ControlSession:
         self.port = sockaddr[1]
         try:
             self.sock = socket.socket(family, socktype, protocol)
-        except socket.error as (errno, msg):
+        except socket.error as e:
             raise ControlException("Error opening %s: %s [%d]" \
-                                   % (hname, msg, errno))
+                                   % (hname, e.msg, e.errno))
         try:
             self.sock.connect(sockaddr)
-        except socket.error as (errno, msg):            
+        except socket.error as e:
             raise ControlException("Error connecting to %s: %s [%d]" \
-                                   % (hname, msg, errno))
+                                   % (hname, e.msg, e.errno))
         return True
 
     def password(self):
-- 
2.7.4

