From d36ff3cb7ecaabc257b201ee444ccc2a15f2396d Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Thu, 1 Dec 2016 21:38:31 -0500
Subject: [PATCH 098/268] Address GitLab issue #177: ntpq > TypeError in
 packet.py

---
 pylib/util.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pylib/util.py b/pylib/util.py
index a2c9999..113cecf 100644
--- a/pylib/util.py
+++ b/pylib/util.py
@@ -80,7 +80,7 @@ def termsize():
             fcntl.ioctl(2, termios.TIOCGWINSZ,
             struct.pack('HHHH', 0, 0, 0, 0)))
         size = (w, h)
-    return TermSize(w, h)
+    return TermSize(*size)
 
 class PeerSummary:
     "Reusable report generator for peer statistics"
-- 
2.7.4

