From 6eb6bc471d6aeaf71dd4c8c9323a89be7803298f Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Fri, 2 Dec 2016 02:45:43 -0500
Subject: [PATCH 112/268] More isolation of the packetstamp code.

---
 ntpd/ntp_io.c          | 6 ------
 ntpd/ntp_packetstamp.c | 2 --
 2 files changed, 8 deletions(-)

diff --git a/ntpd/ntp_io.c b/ntpd/ntp_io.c
index c76f6b8..06ef0f1 100644
--- a/ntpd/ntp_io.c
+++ b/ntpd/ntp_io.c
@@ -10,12 +10,6 @@
 #if !defined(FNM_CASEFOLD) && defined(FNM_IGNORECASE)
 # define FNM_CASEFOLD FNM_IGNORECASE
 #endif
-#ifdef HAVE_SYS_IOCTL_H
-# include <sys/ioctl.h>
-#endif
-#ifdef HAVE_SYS_SOCKIO_H	/* UXPV: SIOC* #defines (Frank Vance <fvance@waii.com>) */
-# include <sys/sockio.h>
-#endif
 #include <sys/uio.h>
 
 #include "ntp_machine.h"
diff --git a/ntpd/ntp_packetstamp.c b/ntpd/ntp_packetstamp.c
index df93eb9..0b8a70a 100644
--- a/ntpd/ntp_packetstamp.c
+++ b/ntpd/ntp_packetstamp.c
@@ -13,10 +13,8 @@
 # include <sys/sockio.h>
 #endif
 
-#include "ntp_machine.h"
 #include "ntpd.h"
 #include "ntp_stdlib.h"
-#include "ntp_assert.h"
 #include "timespecops.h"
 
 #if defined(SO_BINTIME) && defined(SCM_BINTIME) && defined(CMSG_FIRSTHDR)
-- 
2.7.4

