From ed8c489a1185e9a6813eb70aa6bab7c42e06e203 Mon Sep 17 00:00:00 2001
From: Richard Laager <rlaager@wiktel.com>
Date: Fri, 2 Dec 2016 17:15:22 -0600
Subject: [PATCH 120/268] Install ntpq man page

---
 wscript | 1 +
 1 file changed, 1 insertion(+)

diff --git a/wscript b/wscript
index 5ee6e44..10328b5 100644
--- a/wscript
+++ b/wscript
@@ -209,6 +209,7 @@ def build(ctx):
 	ctx.add_post_fun(linkmaker)
 
 	ctx.manpage(8, "ntpleapfetch/ntpleapfetch-man.txt")
+	ctx.manpage(1, "ntpq/ntpq-man.txt")
 	ctx.manpage(1, "ntptrace/ntptrace-man.txt")
 	ctx.manpage(1, "ntpstats/ntpviz-man.txt")
 	ctx.manpage(8, "ntpwait/ntpwait-man.txt")
-- 
2.7.4

