From 5b266de616065a9105f8a8ae595e993280c258f8 Mon Sep 17 00:00:00 2001
From: Matt Selsky <matthew.selsky@twosigma.com>
Date: Fri, 2 Dec 2016 22:59:09 -0500
Subject: [PATCH 126/268] Fix typos in variable names

---
 ntpq/ntpq       | 4 ++--
 pylib/packet.py | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/ntpq/ntpq b/ntpq/ntpq
index 0ad33ec..d421c91 100755
--- a/ntpq/ntpq
+++ b/ntpq/ntpq
@@ -618,7 +618,7 @@ usage: timeout [ msec ]
                         self.say("%s  %s%d\n" % (legend, "mode#", value))
                 else:
                     self.warn("unexpected vc type %s for %s, value %s\n" % (fmt, name, value))
-        except KeyboardOnterrupt:
+        except KeyboardInterrupt:
             self.warn("display interrupted\n")
 
     def do_delay(self, line):
@@ -661,7 +661,7 @@ usage: delay [ msec ]
                     print("current host remains %s" % self.session.hostname)
                 else:
                     print("still no current host")
-            except KeyboardInterrupted:
+            except KeyboardInterrupt:
                 print("lookup interrupted")
 
     def help_host(self):
diff --git a/pylib/packet.py b/pylib/packet.py
index 4ed5b07..764abd8 100644
--- a/pylib/packet.py
+++ b/pylib/packet.py
@@ -547,7 +547,7 @@ class ControlPacket(Packet):
         return True if self.r_e_m_op & 0x80 else False
 
     def is_error(self):
-        return Trye if self.r_e_m_op & 0x40 else False
+        return True if self.r_e_m_op & 0x40 else False
 
     def more(self):
         return True if self.r_e_m_op & 0x20 else False
-- 
2.7.4

