From afdd6f4c4304b9582db599e869c632c4fdf73d3a Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Sat, 3 Dec 2016 00:26:14 -0500
Subject: [PATCH 133/268] Address GitLab issue #183: Bug: ntpsweep traceback

---
 ntpsweep/ntpsweep | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ntpsweep/ntpsweep b/ntpsweep/ntpsweep
index f5cbacd..14cfe6e 100755
--- a/ntpsweep/ntpsweep
+++ b/ntpsweep/ntpsweep
@@ -47,7 +47,7 @@ def scan_host(host, level):
     if host in known_host_info:
         known_host = True
     else:
-        session = ntp.packet.Mode6session()
+        session = ntp.packet.ControlSession()
         session.openhost(host)
         sysvars = session.readvars()
 
-- 
2.7.4

