From df2a28737ea65ea161c12d1fefbb97a55371b1ec Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Sat, 3 Dec 2016 07:51:23 -0500
Subject: [PATCH 138/268] More armoring for termsize().

---
 pylib/util.py | 28 ++++++++++++++++------------
 1 file changed, 16 insertions(+), 12 deletions(-)

diff --git a/pylib/util.py b/pylib/util.py
index 04388eb..aecc3aa 100644
--- a/pylib/util.py
+++ b/pylib/util.py
@@ -69,18 +69,22 @@ TermSize = collections.namedtuple("TermSize", ["width", "height"])
 def termsize():
     "Return the current terminal size."
     # Alternatives at http://stackoverflow.com/questions/566746/how-to-get-console-window-width-in-python
-    if not os.isatty(1):
-        size = (80, 24)
-    try:
-        (w, h) = shutil.get_terminal_size((80, 24))
-        size = (w, h)
-    except AttributeError:
-        # OK, Python version < 3.3, cope
-        import fcntl, termios, struct
-        h, w, hp, wp = struct.unpack('HHHH',
-            fcntl.ioctl(2, termios.TIOCGWINSZ,
-            struct.pack('HHHH', 0, 0, 0, 0)))
-        size = (w, h)
+    # The way this is usedd makes it not a big deal if the default is wrong.
+    size = (80, 24)
+    if os.isatty(1):
+        try:
+            (w, h) = shutil.get_terminal_size((80, 24))
+            size = (w, h)
+        except AttributeError:
+            try:
+                # OK, Python version < 3.3, cope
+                import fcntl, termios, struct
+                h, w, hp, wp = struct.unpack('HHHH',
+                    fcntl.ioctl(2, termios.TIOCGWINSZ,
+                    struct.pack('HHHH', 0, 0, 0, 0)))
+                size = (w, h)
+            except IOError:
+                pass
     return TermSize(*size)
 
 class PeerSummary:
-- 
2.7.4

