From 8cb3177d6d2c3d48ff4c20f004161ae00d0b2f01 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Sat, 3 Dec 2016 09:35:47 -0500
Subject: [PATCH 150/268] Address Gitlab issue #183: Bug: ntpsweep traceback

---
 ntpsweep/ntpsweep | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ntpsweep/ntpsweep b/ntpsweep/ntpsweep
index 14cfe6e..534f71d 100755
--- a/ntpsweep/ntpsweep
+++ b/ntpsweep/ntpsweep
@@ -49,7 +49,7 @@ def scan_host(host, level):
     else:
         session = ntp.packet.ControlSession()
         session.openhost(host)
-        sysvars = session.readvars()
+        sysvars = session.readvar()
 
         # got answers ? If so, go on.
         if isinstance(sysvars, dict):
-- 
2.7.4

