From 552339cf52e55a02ba9e31a945a24018e8588abf Mon Sep 17 00:00:00 2001
From: Matt Selsky <matthew.selsky@twosigma.com>
Date: Sat, 3 Dec 2016 10:29:03 -0500
Subject: [PATCH 152/268] Fix typo in comment

---
 pylib/util.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pylib/util.py b/pylib/util.py
index aecc3aa..a46f66b 100644
--- a/pylib/util.py
+++ b/pylib/util.py
@@ -69,7 +69,7 @@ TermSize = collections.namedtuple("TermSize", ["width", "height"])
 def termsize():
     "Return the current terminal size."
     # Alternatives at http://stackoverflow.com/questions/566746/how-to-get-console-window-width-in-python
-    # The way this is usedd makes it not a big deal if the default is wrong.
+    # The way this is used makes it not a big deal if the default is wrong.
     size = (80, 24)
     if os.isatty(1):
         try:
-- 
2.7.4

