From 18a0a88c3eadc83edb198b1ef1e5927d9e3f2c56 Mon Sep 17 00:00:00 2001
From: Hal Murray <murray@hgm.example.com>
Date: Sun, 4 Dec 2016 03:06:00 -0800
Subject: [PATCH 167/268] Fix for mru avgint < 5 (float/int mixup)

---
 pylib/util.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pylib/util.py b/pylib/util.py
index a46f66b..cebb5dd 100644
--- a/pylib/util.py
+++ b/pylib/util.py
@@ -314,7 +314,7 @@ class MRUSummary:
 
     def summary(self, entry):
         lstint = int(self.now - entry.last + 0.5)
-        active = int(entry.last - entry.first + 0.5)
+        active = float(entry.last - entry.first)
         favgint = active / entry.ct
         avgint = int(favgint + 0.5)
         stats = "%6d" % lstint
-- 
2.7.4

