From 7e4abcffadc8c3737c44859ecd48165dca690f7d Mon Sep 17 00:00:00 2001
From: Chris Mayo <aklhfex@gmail.com>
Date: Sun, 4 Dec 2016 18:24:08 +0000
Subject: [PATCH 171/268] Use https for ntpsec.org

---
 docs/asciidoc.conf | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/docs/asciidoc.conf b/docs/asciidoc.conf
index b1f23da..d4b67b6 100644
--- a/docs/asciidoc.conf
+++ b/docs/asciidoc.conf
@@ -2,10 +2,10 @@
 # and public communications channels indirect through here.
 [attributes]
 project-domain=ntpsec.org
-project-website=http://www.ntpsec.org
-project-weblink=http://www.ntpsec.org[www.ntpsec.org]
+project-website=https://www.ntpsec.org
+project-weblink=https://www.ntpsec.org[www.ntpsec.org]
 project-issuetracker=https://gitlab.com/NTPsec/ntpsec/issues
-project-contact=http://www.ntpsec.org/channels.html
+project-contact=https://www.ntpsec.org/channels.html
 project-security-list=security@ntpsec.org
 project-bug-list=bugs@ntpsec.org
 millshome=http://www.eecis.udel.edu/~mills/
-- 
2.7.4

