From 576c83fc011d4fd27640ed9bf88905b29d0f8d77 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Sun, 4 Dec 2016 10:43:02 -0500
Subject: [PATCH 173/268] Force early creation of generated Python files

---
 pylib/wscript | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pylib/wscript b/pylib/wscript
index fbe49d8..96f88a1 100644
--- a/pylib/wscript
+++ b/pylib/wscript
@@ -37,6 +37,8 @@ def build(ctx):
 	target      = [target3, '../wafhelpers/.autorevision-cache', '../wafhelpers/autorevision.sh'],
 	)
 
+    ctx.add_group() # Force early creation of generated files
+
     ctx(
 	features     ='py',
 	source       = ctx.path.ant_glob('*.py'),
-- 
2.7.4

