From 9195616da06cd3631ca8c29e35f6a4b97dd876e6 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Mon, 5 Dec 2016 06:34:54 -0500
Subject: [PATCH 177/268] Typo fix.

---
 pylib/packet.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pylib/packet.py b/pylib/packet.py
index 764abd8..b3375f0 100644
--- a/pylib/packet.py
+++ b/pylib/packet.py
@@ -373,7 +373,7 @@ class SyncPacket(Packet):
         self.origin_timestamp = 0
         self.receive_timestamp = 0
         self.transmit_timestamp = 0
-        self.data = polybyes(data)
+        self.data = polybytes(data)
         self.extension = ''
         self.extfields = []
         self.mac = ''
-- 
2.7.4

