From 0ebe70f045b82d67e45cb715e9cabb869709fb2f Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Tue, 6 Dec 2016 03:20:37 -0500
Subject: [PATCH 182/268] Fix autobuild of the autorevision cache file.

---
 pylib/wscript | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/pylib/wscript b/pylib/wscript
index 96f88a1..3807dbe 100644
--- a/pylib/wscript
+++ b/pylib/wscript
@@ -12,6 +12,7 @@ def build(ctx):
     target1 = ctx.srcnode.make_node('pylib/ntp_control.py')
     target2 = ctx.srcnode.make_node('pylib/ntp_magic.py')
     target3 = ctx.srcnode.make_node('pylib/version.py')
+    target4 = ctx.srcnode.make_node('wafhelpers/.autorevision-cache')
 
     ctx(
 	before      = ['pyc', 'pyo'],
@@ -32,9 +33,9 @@ def build(ctx):
     ctx(
 	before      = ['pyc', 'pyo'],
 	cwd         = srcnode,
-	rule        = 'VCS_EXTRA=`cat ${SRC}` ../wafhelpers/autorevision.sh -o ${TGT[1].abspath()} -e VERSION -t python >${TGT[0].name}',
-	source      = "../VERSION",
-	target      = [target3, '../wafhelpers/.autorevision-cache', '../wafhelpers/autorevision.sh'],
+	rule        = 'VCS_EXTRA=`cat ${SRC[0]}` ../wafhelpers/autorevision.sh -o ${TGT[1].abspath()} -e VERSION -t python >${TGT[0].name}',
+	source      = ["../VERSION", '../wafhelpers/autorevision.sh'] ,
+	target      = [target3, target4],
 	)
 
     ctx.add_group() # Force early creation of generated files
-- 
2.7.4

