From 6d145723078f3c14e938ddb50ea57e0da26a5262 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Tue, 6 Dec 2016 03:35:43 -0500
Subject: [PATCH 184/268] Fix an incorrect range check.

---
 ntpq/ntpq | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ntpq/ntpq b/ntpq/ntpq
index 9cbb2dc..69be97b 100755
--- a/ntpq/ntpq
+++ b/ntpq/ntpq
@@ -439,7 +439,7 @@ usage: help [ command ]
             return ()
         lo = self.__assoc_valid(tokens[0])
         hi = self.__assoc_valid(tokens[1])
-        if lo < 0 or hi < 0 or hi < 0: 
+        if lo < 0 or hi < 0 or hi < lo: 
             return ()
         return (lo, hi)
 
-- 
2.7.4

