From 48de41aac099c15b431929fcd30d51e7869832b3 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Wed, 7 Dec 2016 06:37:27 -0500
Subject: [PATCH 200/268] Fix nasty typo.

---
 ntpfrob/bumpclock.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ntpfrob/bumpclock.c b/ntpfrob/bumpclock.c
index a6c32ae..573b738 100644
--- a/ntpfrob/bumpclock.c
+++ b/ntpfrob/bumpclock.c
@@ -10,7 +10,7 @@
 #include <string.h>
 #include <time.h>
 
-#define NANOSECONDS	100000000
+#define NANOSECONDS	1000000000
 
 void bumpclock(int bump)
 {
-- 
2.7.4

