From 3eabba194a818368635df14fbc351a85812e7bd4 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Wed, 7 Dec 2016 17:24:16 -0500
Subject: [PATCH 209/268] Correct fatal errors in exception handling.

---
 ntpdig/pyntpdig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ntpdig/pyntpdig b/ntpdig/pyntpdig
index 87481ca..6e46d52 100755
--- a/ntpdig/pyntpdig
+++ b/ntpdig/pyntpdig
@@ -363,7 +363,7 @@ if __name__ == '__main__':
         for server in concurrent_hosts:
             try:
                 returned += queryhost(server=server, concurrent=True, timeout=timeout)
-            except SyncException:
+            except ntp.packet.SyncException as e:
                 log(e.message)
                 continue
             if len(returned) >= samples:
@@ -371,7 +371,7 @@ if __name__ == '__main__':
         for server in arguments:
             try:
                 returned += queryhost(server=server, concurrent=False, timeout=timeout)
-            except SyncException:
+            except ntp.packet.SyncException as e:
                 log(e.message)
                 continue
             if len(returned) >= samples:
-- 
2.7.4

