From 41598f045db3ad6a781d7513e19f7e747c844918 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Wed, 7 Dec 2016 20:48:01 -0500
Subject: [PATCH 214/268] NEWS update.

---
 NEWS | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/NEWS b/NEWS
index 9671bf2..fbee92b 100644
--- a/NEWS
+++ b/NEWS
@@ -12,7 +12,15 @@ ntpkeygen has been moved from C to Python.  This is not a functional
 change, just another move to improve maintainability and reduce attack
 surface by decreasing line count.
 
-A new tool for time-=service operators, ntpmon, supports real-time
+ntpdig has also been moved from C to Python. Though this is also
+mostly a move to reduce line count, the new version does have some
+functional changes.  Obsolete options have been dropped, logging is
+done a bit differently, and the synchronization-distance computation has
+been brought up to date with ntpd's. Also, this version can be told to
+collect multiple samples and use whichever has the lowest combination
+of stratum and synch distance.
+
+A new tool for time-service operators, ntpmon, supports real-time
 monitoring of your NTP installation's status.
 
 == 2016-11-23: 0.9.5 ==
-- 
2.7.4

