From 3a4a0ac076c625ff825948a653df7c5d286855e8 Mon Sep 17 00:00:00 2001
From: Matt Selsky <matthew.selsky@twosigma.com>
Date: Thu, 8 Dec 2016 00:46:55 -0500
Subject: [PATCH 224/268] ntpdig doesn't use warn() but sys.stderr.write()

---
 ntpclients/ntpdig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ntpclients/ntpdig b/ntpclients/ntpdig
index 0805477..84b0f9a 100755
--- a/ntpclients/ntpdig
+++ b/ntpclients/ntpdig
@@ -341,7 +341,7 @@ if __name__ == '__main__':
             raise SystemExit(1)
 
     if not credentials and authkey and keyfile is None:
-        self.warn("-a option requires -k.\n")
+        sys.stderr.write("-a option requires -k.\n")
         raise SystemExit(1)
 
     gap /= 1000	# Scale gap to milliseconds
-- 
2.7.4

