From 040c4fa8840a4f3e3a7c5011b7605849c86d5e95 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Thu, 8 Dec 2016 17:30:42 -0500
Subject: [PATCH 231/268] Reverse order of span entries in ntpmon MRU display.

---
 ntpclients/ntpmon | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ntpclients/ntpmon b/ntpclients/ntpmon
index e178a55..8532b47 100755
--- a/ntpclients/ntpmon
+++ b/ntpclients/ntpmon
@@ -121,7 +121,7 @@ if __name__ == '__main__':
                     if span.entries:
                         stdscr.addstr(ntp.util.MRUSummary.header + "\n",
                                       curses.A_BOLD)
-                        for entry in span.entries:
+                        for entry in reversed(span.entries):
                             stdscr.addstr(mru_report.summary(entry) + "\n")
                 except curses.error:
                     # An addstr overran the screen, no worries
-- 
2.7.4

