From 4226b067ca1a0165a3fd24d63943058411872c62 Mon Sep 17 00:00:00 2001
From: Matt Selsky <matthew.selsky@twosigma.com>
Date: Thu, 8 Dec 2016 23:28:03 -0500
Subject: [PATCH 236/268] Typo

---
 pylib/util.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pylib/util.py b/pylib/util.py
index 2233a74..3a7494d 100644
--- a/pylib/util.py
+++ b/pylib/util.py
@@ -45,7 +45,7 @@ def portsplit(hostname):
             hostname = hostname[1:-1]	# Strip brackets
     return (hostname, portsuffix)
 
-# A hack to avoid repleatedly hammering on DNS when ntpmon runs.
+# A hack to avoid repeatedly hammering on DNS when ntpmon runs.
 canonicalization_cache = {}
 
 def canonicalize_dns(inhost, family=socket.AF_UNSPEC):
-- 
2.7.4

