From f31ec4bbd6e819c0fe4632d35fb2a293b19399d5 Mon Sep 17 00:00:00 2001
From: Matt Selsky <matthew.selsky@twosigma.com>
Date: Thu, 8 Dec 2016 23:55:03 -0500
Subject: [PATCH 239/268] Remove double "import curses" from ntpmon

---
 ntpclients/ntpmon | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ntpclients/ntpmon b/ntpclients/ntpmon
index 22141e5..a6d1cb4 100755
--- a/ntpclients/ntpmon
+++ b/ntpclients/ntpmon
@@ -5,7 +5,7 @@
 # SPDX-License-Identifier: BSD-2-clause
 from __future__ import print_function, division
 
-import sys, curses, time, getopt
+import sys, time, getopt
 
 try:
     import ntp.packet
-- 
2.7.4

