From f105623eab3da44008c78d5e6e9ecb5eccc00c90 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Fri, 9 Dec 2016 20:17:53 -0500
Subject: [PATCH 255/268] Port fix for glibc versions < 2.12.

---
 ntptime/ntptime.c       | 5 +++--
 wafhelpers/configure.py | 2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/ntptime/ntptime.c b/ntptime/ntptime.c
index 9689684..e4730b2 100644
--- a/ntptime/ntptime.c
+++ b/ntptime/ntptime.c
@@ -353,12 +353,13 @@ main(
 			       (u_int)ts.l_ui, (u_int)ts.l_uf,
 			       (int)ntv.time.tv_sec, fdigits,
 			       (int)time_frac,
+
 			       ctime_r((time_t *)&ntv.time.tv_sec, ascbuf));
-#if NTP_API > 3
+#if defined(STRUCT_NTPTIMEVAL_HAS_TAI) && NTP_API > 3
 		printf(json ? jfmt5 : ofmt5, (long)ntv.tai);
 #else
 		printf(json ? jfmt6 : ofmt6);
-#endif /* NTP_API */
+#endif /* defined(STRUCT_NTPTIMEVAL_HAS_TAI) && NTP_API */
 	}
 	status = ntp_adjtime_ns(&ntx);
 	if (status < 0) {
diff --git a/wafhelpers/configure.py b/wafhelpers/configure.py
index d2c0c70..0f22154 100644
--- a/wafhelpers/configure.py
+++ b/wafhelpers/configure.py
@@ -259,7 +259,7 @@ def cmd_configure(ctx, config):
                 ("time_tick", "timex", ["sys/time.h", "sys/timex.h"]),
                 ("modes", "timex", ["sys/time.h", "sys/timex.h"]),
                 ("time.tv_nsec", "ntptimeval", ["sys/time.h", "sys/timex.h"]),
-                ("tai", "ntptimeval", ["sys/time.h", "sys/timex.h"]),
+                ("tai", "ntptimeval", ["sys/time.h", "sys/timex.h"]), # first in glibc 2.12
         )
         for (f, s, h) in structure_fields:
                 check_structfield(ctx, f, s, h)
-- 
2.7.4

