From 0014d865229f47cd568052478418788c2b1f42b1 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Fri, 9 Dec 2016 23:37:44 -0500
Subject: [PATCH 257/268] In ntpmon, show the server version in the status
 line.

---
 ntpclients/ntpmon | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ntpclients/ntpmon b/ntpclients/ntpmon
index c62ae80..92542fb 100755
--- a/ntpclients/ntpmon
+++ b/ntpclients/ntpmon
@@ -36,7 +36,7 @@ def iso8601(t):
 def statline(_peerlist, _mrulist):
     "Generate a status line"
     # We don't use stdversion here because the presence of a date is confusing
-    leader = "NTPsec %s-%s" % (ntp.version.VERSION, ntp.version.VCS_TICK)
+    leader = sysvars['version']
     trailer = "Last update: %s" % iso8601(span.entries[0].last)
     spacer = (peer_report.termwidth - len(leader) - len(trailer)) * " "
     return leader + spacer + trailer
@@ -98,6 +98,7 @@ if __name__ == '__main__':
     try:
         session = ntp.packet.ControlSession()
         session.openhost(arguments[0] if arguments else "localhost")
+        sysvars = session.readvar()
         with OutputContext() as ctx:
             while True:
                 try:
-- 
2.7.4

