From e2ce5325be9535c5a35e8ff3f0b698d0668e8c7a Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Sat, 10 Dec 2016 17:50:00 -0500
Subject: [PATCH 268/268] Remove dead code.

---
 ntpd/ntp_io.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/ntpd/ntp_io.c b/ntpd/ntp_io.c
index 7b8c192..1d94264 100644
--- a/ntpd/ntp_io.c
+++ b/ntpd/ntp_io.c
@@ -228,8 +228,6 @@ struct remaddr {
 
 remaddr_t *	remoteaddr_list;
 endpt *		ep_list;	/* complete endpt list */
-endpt *		mc4_list;	/* IPv4 mcast-capable unicast endpts */
-endpt *		mc6_list;	/* IPv6 mcast-capable unicast endpts */
 
 static endpt *	wildipv4;
 static endpt *	wildipv6;
-- 
2.7.4

