Origin: upstream, https://gitlab.com/NTPsec/ntpsec/commit/acb2ecdcabad2ab42e9c6352999e174dd102eb3f
Bug: https://gitlab.com/NTPsec/ntpsec/issues/509
From: "Gary E. Miller" <gem@rellim.com>
Date: Mon, 22 Oct 2018 13:47:20 -0700
Subject: [PATCH] ntp_control: Stop potentially dereferencing a NULL pointer.

---
 ntpd/ntp_control.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/ntpd/ntp_control.c
+++ b/ntpd/ntp_control.c
@@ -2888,6 +2888,7 @@
 	const struct ctl_var *v;
 	int ext_var;
 	char *valuep;
+	char nulltxt[1] = { '\0' };
 	long val;
 	size_t octets;
 	char *vareqv;
@@ -2933,6 +2934,9 @@
 			ctl_error(CERR_PERMISSION);
 			return;
 		}
+		if (NULL == valuep)
+			valuep = nulltxt;
+
 		errno = 0;
 		if (!ext_var && (*valuep == '\0'
 				 || (val = strtol(valuep, NULL, 10), errno != 0))) {
