Origin: upstream, https://gitlab.com/NTPsec/ntpsec/commit/2ccb87ab78d7ca962f8a903499a4e50a379c56dd
Bug: https://gitlab.com/NTPsec/ntpsec/issues/507
Bug: https://gitlab.com/NTPsec/ntpsec/issues/508
From: "Gary E. Miller" <gem@rellim.com>
Date: Wed, 17 Oct 2018 20:55:06 -0700
Subject: [PATCH] ntp_control: Check for count too big in control packet.

---
 ntpd/ntp_control.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

--- a/ntpd/ntp_control.c
+++ b/ntpd/ntp_control.c
@@ -825,6 +825,13 @@
 		return;
 	}
 
+	if (CTL_MAX_DATA_LEN < req_count) {
+                /* count to big */
+		ctl_error(CERR_BADFMT);
+		numctlbadpkts++;
+		return;
+	}
+
 	properlen = req_count + (int)CTL_HEADER_LEN;
 	/* round up proper len to a 8 octet boundary */
 
@@ -2505,7 +2512,7 @@
 
 	/* Scan the string in the packet until we hit comma or
 	 * EoB. Register position of first '=' on the fly. */
-	for (tp = NULL, cp = reqpt; cp != reqend; ++cp) {
+	for (tp = NULL, cp = reqpt; cp < reqend; ++cp) {
 		if (*cp == '=' && tp == NULL)
 			tp = cp;
 		if (*cp == ',')
