Description: Make ntpviz optionfiles consistent
Forwarded: no
Origin: vendor
Author: Richard Laager <rlaager@wiktel.com>
Last-Update: 2016-12-19
--- a/www/day/optionfile
+++ b/www/day/optionfile
@@ -1,6 +1,4 @@
-# this is a comment
---name=ntpvis-day # another comment
---outdir day --period=1
+--outdir day
+--period=1
 # uncomment next line to clip the data
 #--clip
-
--- a/www/week/optionfile
+++ b/www/week/optionfile
@@ -1,5 +1,3 @@
-# this is a comment
---name=ntpvis-week # another comment
 --outdir=week
 --period=7
 # uncomment next line to clip the data
